test(minifier): add and enable some tests in fold_constants

sapphi-red opened the pull request:
Ported some tests that was missing from closure compiler.
Also enabled some tests.

sapphi-red commented:
* **#8769** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8769?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8769?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8769).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

sapphi-red commented on crates/oxc_minifier/src/peephole/fold_constants.rs:
```diff
@@ -1482,39 +1513,45 @@ mod test {
         fold("8589934591 >>> 0", "4294967295");
         fold("8589934592 >>> 0", "0");
         fold("8589934593 >>> 0", "1");
+
+        fold("x = -1 << 1", "x = -2");
+        fold("x = -1 << 8", "x = -256");
+        fold("x = -1 >> 1", "x = -1");
+        fold("x = -2 >> 1", "x = -1");
+        fold("x = -1 >> 0", "x = -1");
     }
 
     #[test]
     fn test_string_add() {
         fold("x = 'a' + 'bc'", "x = 'abc'");
         fold("x = 'a' + 5", "x = 'a5'");
         fold("x = 5 + 'a'", "x = '5a'");
-        // test("x = 'a' + 5n", "x = 'a5n'");
-        // test("x = 5n + 'a'", "x = '5na'");
+        fold("x = 'a' + 5n", "x = 'a5'");
+        fold("x = 5n + 'a'", "x = '5a'");
```
The original test case was incorrect.

Boshen commented:
### Merge activity

* **Jan 28, 7:16 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
test(minifier): add and enable some tests in fold_constants (#8769)

Ported some tests that was missing from closure compiler.
Also enabled some tests.
